packages/shared/components/atoms/pills/ActivityTypePill.svelte

```diff
@@ -0,0 +1,68 @@
+<script lang="typescript">
+    import { localize } from '@core/i18n'
+    import { isParticipationPayload } from '@lib/wallet'
+    import Pill from './Pill.svelte'
+
+    export let activity
+    export let confirmed
+
+    enum ActivityType {
+        StakingTransaction = 'stakingTransaction',
+        Transfer = 'transfer',
+        Transferring = 'transferring',
+        Received = 'received',
+        Sent = 'sent',
+        Receiving = 'receiving',
+        Sending = 'sending',
+    }
+
+    let type: ActivityType
+    $: {
+        if (activity) {
+            if (isParticipationPayload(activity)) {
```

Comment:
It seems weird to be passing an `activity` to the `isParticipationPayload` function imo. Perhaps there is a better / more descriptive word than `activity`?

Reply:
Yeah its because I was renaming things to activity within this portion of code, because this is part of the ActivityDetailPopop openable by a ActivityCard. Also for now just trying to update UI and maintain backwards compatibility with existing IOTA before we refactor some of the logic for stardust.

Reply:
That makes sense, although maybe we can rename the function in that case?

Reply:
Stardust will not have the participations for now. Would be good to refactor when its added. I imagine in a few PRs time I will remove this function call for now 👍